hipchat: Fix user notification for v2 api

hostmaster opened the pull request:
`notify` parameter is not working as expected for hipchat API v2. Parameter `notify: no` enables user notification and vise versa. In fact user notification is disabled by default. 

I think there is a little misunderstanding here. [Send message](https://www.hipchat.com/docs/apiv2/method/send_message) request is not supposed to be an equivalent of [room/message](https://www.hipchat.com/docs/api/method/rooms/message) from v1.It simply does not support all the parameters. [Send room notification](https://www.hipchat.com/docs/apiv2/method/send_room_notification) is a right one. It supports `notify` parameter to configure a user notification.

pb8226 commented:
Thanks for the changes. Do you have a playbook that you used for testing? I want to make sure I test the scenario you're fixing.

hostmaster commented:
I don't think there is a scenaruio. I mean I used version from https://github.com/ansible/ansible-modules-extras/pull/118 before and then I just updated the module from upstream and found that `notify` is disabled now.

I'm using `api: https://api.hipchat.com/v2/`

hostmaster commented:
@pb8226 however I think i can create something like that if it helps.

pb8226 commented:
@hostmaster Please if you can, no rush obviously.  I have a playbook I used for testing but it obviously didn't capture what you're testing so I wanted to run my playbook and then yours to see where the difference was.

hostmaster commented:
@pb8226 ok so just to clarify. you are using hipchat api v2 and user notification is working as expected right ?

pb8226 commented:
This was the playbook I used for testing:

https://www.dropbox.com/s/sk2580g7j0iv56a/test-playbook.yml?dl=0

bcoca commented:
@pb8226 is this OK to merge? @shirou can you also review?

pb8226 commented:
@bcoca I was waiting for a playbook from @hostmaster that showed the bug he was fixing.  Notifications were working for me with the previous code and I provided him the playbook I used for testing.  I haven't heard back after providing the playbook to him.

hostmaster commented:
@pb8226 ok I have  created  a slightly modified version of your playbook (I put hipchat settings in a separate file ) https://gist.github.com/hostmaster/3d554473be59bfc75894

hostmaster commented:
@pb8226 Here is my result

```
$ ansible-playbook -i localhost test_playbook.yml

PLAY [Test HipChat Module] ****************************************************

TASK: [Notify Room V2] ********************************************************
changed: [localhost]
```

Message is sent, **no notification**, color red 

```
TASK: [Notify Room V2] ********************************************************
changed: [localhost]
```

Message is sent, **no notification**, color purple

```
TASK: [Message Room V2] *******************************************************
changed: [localhost]
```

Message is sent, **user notification**, no color

```
TASK: [Notify Room V1] ********************************************************
changed: [localhost]

TASK: [Notify Room V1] ********************************************************
changed: [localhost]

TASK: [Message Room V1] *******************************************************
changed: [localhost]

PLAY RECAP ********************************************************************
localhost                  : ok=6    changed=6    unreachable=0    failed=0
```

hostmaster commented:
@pb8226 `notify` parameter is optional the case when it is ommited is not covered in your playbook. In my test it doesn't make user notification for API v2. (it should)

pb8226 commented:
@hostmaster ok thank you, that makes sense. @bcoca this can be merged.

shirou commented:
FYI: I don't use hipchat anymore, I can not test. But it looks good for me. Thank you for updating!

gregdek commented:
Thanks @shirou.

@pb8226 would you be interested in taking over maintainership of this module?

hostmaster commented:
I've just received a clarification from Hipchat support regarding the difference among `Send Message` and `Send Room notification` API calls. 

>    Send Room Notification - Provides the user with more options to configure how the API is being sent out with different content-types, bigger request body, accessible by.
>    Send Message : Simple and straight forward. Only accessible by users and it only has a single request body which is message.
> 
> It depends really how you want to make use of for your integration. If you want more flexibility, you can se /notification. If you want something which is simple and lightweight, you can make use of /message.
